src/main.ts

```diff
@@ -1,144 +1,56 @@
+      },
+      'SOL',
+    ),
+  ]));
+  for (const [symbol, router] of routers) {
+    if (productStatus.get(symbol)) {
+      await router.refresh_dips();
+      console.log(`Check ${symbol} Position vs MM Quotes ${new Date().toUTCString()}`);
+      router.checkMMPrices();
+      await sleepExact(productStaggerSec);
+    }
   }
 
   setInterval(async () => {
     try {
-      if (productStatus.get('SOL')) {
-        console.log('------------------------------------------------');
-        console.log('RERUN SOL Risk Manager', new Date().toUTCString());
-        await solRouter.refresh_dips();
-        console.log('Re-Check SOL Position vs MM Quotes', new Date().toUTCString());
-        solRouter.checkMMPrices();
-        await sleepExact(productStaggerSec);
-      }
-      if (productStatus.get('BTC')) {
-        console.log('------------------------------------------------');
-        console.log('RERUN BTC Risk Manager', new Date().toUTCString());
-        await btcRouter.refresh_dips();
-        console.log('Re-Check BTC Position vs MM Quotes', new Date().toUTCString());
-        btcRouter.checkMMPrices();
-        await sleepExact(productStaggerSec);
-      }
-      if (productStatus.get('ETH')) {
-        console.log('------------------------------------------------');
-        console.log('RERUN ETH Risk Manager', new Date().toUTCString());
-        await ethRouter.refresh_dips();
-        console.log('Re-Check ETH Position vs MM Quotes', new Date().toUTCString());
-        ethRouter.checkMMPrices();
-        await sleepExact(productStaggerSec);
-      }
-      if (productStatus.get('BONK')) {
-        console.log('------------------------------------------------');
-        console.log('RERUN BONK Risk Manager', new Date().toUTCString());
-        await bonkRouter.refresh_dips();
-        console.log('Re-Check BONK Position vs MM Quotes', new Date().toUTCString());
-        bonkRouter.checkMMPrices();
-        await sleepExact(productStaggerSec);
-      }
-      if (productStatus.get('MNGO')) {
-        console.log('------------------------------------------------');
-        console.log('RERUN MNGO Risk Manager', new Date().toUTCString());
-        await mngoRouter.refresh_dips();
-        console.log('Re-Check MNGO Position vs MM Quotes', new Date().toUTCString());
-        mngoRouter.checkMMPrices();
+      for (const [symbol, router] of routers) {
+        if (productStatus.get(symbol)) {
+          console.log('------------------------------------------------');
+          console.log(`RERUN ${symbol} Risk Manager ${new Date().toUTCString()}`);
+          await router.refresh_dips();
+          console.log(`Re-Check ${symbol} Position vs MM Quotes ${new Date().toUTCString()}`);
+          router.checkMMPrices();
+          await sleepExact(productStaggerSec);
+        }
       }
     } catch (err) {
       console.log(err);
       console.log(err.stack);
     }
-  }, 1_000 * (scalperWindowSec + (((Math.random() * 2) - 1)
```

Comment:
Unless you think its bad for performance. This was a safety to avoid being front run/predictability of the delta hedges

Reply:
Fine by me then, I think there are enough async's asycn's littered that exact order timing is still relatively unpredictable externally